src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java

```diff
@@ -199,6 +208,7 @@ public final void traverse() {
             // Since reads in each shard are lazily fetched, we need to pass the filter to the window
             // instead of filtering the reads directly here
             readShard.setReadFilter(countedFilter);
+            readShard.setDownsampler(maxReadsPerAlignmentStart > 0 ? new PositionalDownsampler(maxReadsPerAlignmentStart, getHeaderForReads()) : null);
```

Comment:
There already is a no-op downsampler (`PassThroughDownsampler`), but I decided not to use it since it adds a bit of overhead (one `List` creation per start position).